common/persistence/nexus_service_manager.go

```diff
@@ -48,16 +52,20 @@ var (
 type (
 	nexusServiceManagerImpl struct {
 		persistence NexusServiceStore
+		serializer  serialization.Serializer
+		logger      log.Logger
 	}
 )
 
-var _ NexusServiceManager = (*nexusServiceManagerImpl)(nil)
```

Comment:
This should have incoming in the name, we'll want to distinguish between incoming and outgoing everywhere in the codebase.